controllers/external-accounts.js

```diff
@@ -41,4 +44,32 @@ const getExternalAccountData = async (req, res) => {
   }
 };
 
-module.exports = { addExternalAccountData, getExternalAccountData };
+const syncExternalAccountData = async (req, res) => {
+  try {
+    const discordUserData = await getDiscordMembers();
+    const rdsUserData = await getDiscordUsers();
+    rdsUserData.forEach(async (rdsUser) => {
+      discordUserData.forEach(async (discordUser) => {
+        if (rdsUser.discordId === discordUser.user.id) {
+          const userData = {
+            discordJoinedAt: discordUser.joined_at,
+            roles: {
+              ...rdsUser.roles,
+              in_discord: true,
+            },
+          };
+          await addOrUpdate(userData, rdsUser.id);
+        }
+      });
```

Comment:
- Rather than have two for each loop a cleaner way could have been we filter out the users from `rdsUserdata` if they exist in `discordUserData` which would give us the required users to then run the `addOrUpdate` operation on, what do you think?

Reply:
The data is already filtered, so basically the outer loop has less data than the inner loop, the rdsUserData is the users who have run the `/verify` command. So, I don't feel the need to filter further.

Reply:
Update the code using 2 loops and a map reducing the complexity from o(n^2) to o(m+n).
In the process, I have created a map that uses extra memory. Are we ok with the tradeoff?